library/src/test/java/com/vinted/coper/CoperImplTest.kt

```diff
@@ -686,14 +732,40 @@ class CoperImplTest {
         }
     }
 
+    @Test(expected = TimeoutCancellationException::class)
+    fun getFragmentSafely_lifecycleNotReady_crashWithTimeout() {
+        runBlocking {
+            val activityController = Robolectric.buildActivity(FragmentActivity::class.java)
+            val activity = spy(activityController.get())
+
+            val fixture = getCoperInstance(
+                fragmentManager = activity.supportFragmentManager,
+                lifecycle = activity.lifecycle,
+            )
+
+            withTimeout(100) {
+                fixture.getFragmentSafely()
+            }
+        }
+    }
```

Comment:
I suppose you removed such case

Reply:
I added as example, but probably its notplace for that purpose

Reply:
Oh, now I see. Then a test may be good. Just I am wondering, does not this test cause flakiness? I do not see any control of activity in the test. BTW, why are use `spy` everywhere?  `spy` was designed to deal with legacy code and it is very strange to see new code dealing with legacy. Also, keep in mind `activityController` will use its own instance, not spied one.

Reply:
I removed spy here, I use spy, cuase i need to mock one permission method sometimes. Not in all methods I need this, but sometimes I just copy init params, so this was that case, I removed. One case is :
```
whenever(activity.checkPermission(eq(permission), anyOrNull(), anyOrNull()))
            .thenReturn(result)
```

Reply:
I found `checkPermission` is not final. It means you could create your own activity which could support better support of permission handling for tests.

Reply:
I would create separate PR for moving everything to new testing solutions

Reply:
https://github.com/vinted/coper/issues/7